src/core/main.c

```diff
@@ -3045,13 +3052,14 @@ int main(int argc, char *argv[]) {
         mac_selinux_finish();
 
         if (IN_SET(r, MANAGER_REEXECUTE, MANAGER_SWITCH_ROOT))
-                do_reexecute(argc, argv,
-                             &saved_rlimit_nofile,
-                             &saved_rlimit_memlock,
-                             fds,
-                             switch_root_dir,
-                             switch_root_init,
-                             &error_message); /* This only returns if reexecution failed */
+                r = do_reexecute(r,
+                                 argc, argv,
+                                 &saved_rlimit_nofile,
+                                 &saved_rlimit_memlock,
+                                 fds,
+                                 switch_root_dir,
+                                 switch_root_init,
+                                 &error_message); /* This only returns if reexecution failed */
```

Comment:
Same here. Is it not necessary to convert negative `r` into `retval = EXIT_FAILURE` or something?

Reply:
retval is initalized to `EXIT_FAILURE`. If `r < 0`, we never set it to anything else, so we should end up either freezing (for PID1) or returning EXIT_FAILURE.